### PR TITLE
Modules: refactor module classes naming and namespace

### PR DESCRIPTION
**Pull request by yookoala**
**Refactor**

## Description
* Rename `Module\*` class as `Gibbon\Module\*` for more clarity.
* Follow PSR1 class naming (CamelCase) standard.

## Motivation and Context

In principle,
1. All modules' class should stay in namespace "Gibbon\Module\{module name}".
2. All modules' class file should be in "modules\{module name}\src" folder.
3. All modules' class name should be in "CamelCase" format (begins with capital letter).

Namespace of all class of the Gibbon project should be within the `Gibbon` organization namespace. Its a common practice for huge projects with modules. That includes the core modules. The previously used namespace `Module` is ambiguous.

PSR1 naming suggest to use `CamelCase` for class name. And the usual practice would be to use `$camelCase` as default naming of an instance.

## How Has This Been Tested?
* Travis CI test passed.
* Brief test in local installation. (Thou, because I do not know how to access some of them, not all path was tested.)

### CODE REVIEW

**Comment by SKuipers** [comment]
Hi Koala. I'll take a look and review this. I see you've pushed some changes to the Module class and autoloading, which extends outside the scope of this refactoring, and it is already a fairly large PR. Can you separate those out to keep this PR self-contained? Thanks!

**Comment by yookoala** [comment]
@SKuipers: Sure :-)

**Comment by yookoala** [comment]
Also removed the composer file change (i.e. `classmap` autoloading).

File: modules/Attendance/attendance_take_byRollGroupProcess.php [hunk 1]
```diff
@@ -17,11 +17,13 @@
...
 
+use Gibbon\Module\Attendance\AttendanceView;
+
 //Gibbon system-wide includes
-include '../../gibbon.php';
+require __DIR__ . '/../../gibbon.php';
 
 //Module includes
-include './moduleFunctions.php';
+require_once '/moduleFunctions.php';
```
Comment (SKuipers) [inline modules/Attendance/attendance_take_byRollGroupProcess.php]:
```suggestion
require_once __DIR__ .  '/moduleFunctions.php';
```

File: gibbon.php [hunk 2]
```diff
@@ -47,7 +47,7 @@
 // Autoload the current module namespace
 if (!empty($gibbon->session->get('module'))) {
     $moduleNamespace = preg_replace('/[^a-zA-Z0-9]/', '', $gibbon->session->get('module'));
-    $autoloader->addPsr4('Gibbon\\'.$moduleNamespace.'\\', realpath(__DIR__).'/modules/'.$gibbon->session->get('module'));
+    $autoloader->addPsr4('Gibbon\\'.$moduleNamespace.'\\', realpath(__DIR__).'/modules/'.$gibbon->session->get('module').'/src');
```
Comment (SKuipers) [inline gibbon.php]:
```suggestion
    $autoloader->addPsr4('Gibbon\\Module\\'.$moduleNamespace.'\\', realpath(__DIR__).'/modules/'.$gibbon->session->get('module').'/src');
```

File: cli/attendance_weeklySummaryEmail.php [hunk 3]
```diff
@@ -31,9 +32,9 @@
 } else {
     setCurrentSchoolYear($guid, $connection2);
 
-    require_once $_SESSION[$guid]['absolutePath'].'/modules/Attendance/moduleFunctions.php';
-    require_once $_SESSION[$guid]['absolutePath'].'/modules/Attendance/src/attendanceView.php';
-    $attendance = new Module\Attendance\attendanceView($gibbon, $pdo);
+    require_once __DIR__ . '/modules/Attendance/moduleFunctions.php';
```
Comment (SKuipers) [inline cli/attendance_weeklySummaryEmail.php]: __DIR__ is in the cli directory in this instance
```suggestion
    require_once __DIR__ . '/../modules/Attendance/moduleFunctions.php';
```

File: cli/attendance_weeklySummaryEmail.php [hunk 4]
```diff
@@ -31,9 +32,9 @@
...
+    require_once __DIR__ . '/modules/Attendance/src/AttendanceView.php';
```
Comment (SKuipers) [inline cli/attendance_weeklySummaryEmail.php]:
```suggestion
    require_once __DIR__ . '/../modules/Attendance/src/AttendanceView.php';
```

File: modules/Attendance/report_graph_byType.php [hunk 5]
```diff
@@ -80,8 +80,8 @@ function getDateRange($guid, $connection2, $first, $last, $step = '+1 day', $out
     $rollGroups = !empty($_POST['gibbonRollGroupID'])? $_POST['gibbonRollGroupID'] : array('all');
     if (in_array('all', $rollGroups)) $rollGroups = array('all');
 
-    require_once $_SESSION[$guid]['absolutePath'].'/modules/Attendance/src/attendanceView.php';
-    $attendance = new Module\Attendance\attendanceView($gibbon, $pdo);
+    require_once __DIR__ . '/src/AttendanceView.php';
+    $attendance = new AttendanceView($gibbon, $pdo);
```
Comment (SKuipers) [inline modules/Attendance/report_graph_byType.php]: Needs a `namespace Gibbon\Module\Attendance;`

File: cli/attendance_weeklySummaryEmail.php (hunk 3 above)
Comment (yookoala) [reply cli/attendance_weeklySummaryEmail.php]: "moduleFunctions.php" is obsoleted here?

File: modules/Attendance/report_graph_byType.php (hunk 5 above)
Comment (yookoala) [reply modules/Attendance/report_graph_byType.php]: Will add `use` statement on top for this.

**Comment by SKuipers** [comment]
This is looking good. I'll merge and do one more local test. Thanks!!

**Pull request merged**